tests/mockServer/backendSynchronizator.js

```diff
@@ -0,0 +1,413 @@
+    await fetchLogs(chunks[i])
+  }
+
+  return responseData
+}
+
+const fetchLog = async (host, projectName, uid) => {
+  try {
+    const response = await axios.get(`${host}log/${projectName}/${uid}`)
+
+    return response.data
+  } catch (e) {
+    if (e.config && e.config.url) {
+      console.log(e.config.url)
+    } else {
+      console.log(e)
+    }
+  }
+}
+
+const synchronizeBackend = async () => {
+  const { projects } = await fetchData(baseMlRunUrl, 'projects')
+  const projectNames = projects.map(project => project.metadata.name)
+
+  const { project_summaries } = await fetchData(
+    baseMlRunUrl,
+    'project-summaries'
+  )
+  const summaries = { projects: project_summaries }
+
+  const frontendSpec = await fetchData(baseMlRunUrl, 'frontend-spec')
+  const artifactsArr = await fetchJsons(
+    projectNames,
+    baseMlRunUrl,
+    'artifacts?project=',
+    '&tag=*'
+  )
+  const artifacts = convertFromArrayToJson(artifactsArr)
+
+  const featureSetsArr = await fetchJsons(
+    projectNames,
+    baseMlRunUrl,
+    'projects/',
+    '/feature-sets'
+  )
+  const featureSets = convertFromArrayToJson(featureSetsArr)
+
+  const schedulesArr = await fetchJsons(
+    projectNames,
+    baseMlRunUrl,
+    'projects/',
+    '/schedules'
+  )
+  const schedules = convertFromArrayToJson(schedulesArr)
+
+  const featuresArr = await fetchJsons(
+    projectNames,
+    baseMlRunUrl,
+    'projects/',
+    '/features'
+  )
+  const features = convertFromArrayToJson(featuresArr)
+
+  const entitiesArr = await fetchJsons(
+    projectNames,
+    baseMlRunUrl,
+    'projects/',
+    '/entities'
+  )
+  const entities = convertFromArrayToJson(entitiesArr)
+
+  const featureVectorsArr = await fetchJsons(
+    projectNames,
+    baseMlRunUrl,
+    'projects/',
+    '/feature-vectors'
+  )
+  const featureVectors = convertFromArrayToJson(featureVectorsArr)
+
+  const artifactTags = await fetchJsons(
+    projectNames,
+    baseMlRunUrl,
+    'projects/',
+    '/artifact-tags'
+  )
+  const functionsArr = await fetchJsons(
+    projectNames,
+    baseMlRunUrl,
+    'funcs?project='
+  )
+  const functions = convertFromArrayToJson(functionsArr)
+
+  const runsData = await fetchJsons(projectNames, baseMlRunUrl, 'runs?project=')
+  const runs = convertFromArrayToJson(runsData)
+  const runsUid = convertRuns(runs)
+  const runProjectUidData = await fetchRuns(runsUid, baseMlRunUrl, 'run/', '/')
+  const runProjectUid = convertFromArrayToJson(runProjectUidData)
+
+  const pipelines = await fetchJsonsPerProject(
+    projectNames,
+    baseMlRunUrl,
+    'projects/',
+    '/pipelines'
+  )
+  const pipelineIds = getPipelineIds(pipelines)
+  const pipelineIdsData = await fetchJsons(
+    pipelineIds,
+    baseMlRunUrl,
+    'pipelines/'
+  )
+
+  const artifactsLogs = await fetchArtifactsLogs(runs.runs)
+  const logs = await fetchAllLogs(artifactsLogs)
+
+  await clearDataFolder(saveFolder)
+
+  saveDataToJson('./data/projects.json', projects)
```

Comment:
That line saves an only a list of projects, following variant works better `saveDataToJson('./data/projects.json', { 'projects': projects })`